pkg/ring/kv/memberlist/kv_init_service.go

```diff
@@ -86,64 +87,103 @@ func (kvs *KVInitService) stopping(_ error) error {
 
 func (kvs *KVInitService) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	kv := kvs.getKV()
-	var ml *memberlist.Memberlist
-	var store map[string]valueDesc
-
-	if kv != nil {
-		ml = kv.memberlist
-		store = kv.storeCopy()
+	if kv == nil {
+		http.Error(w, "This Cortex instance doesn't use memberlist.", http.StatusOK)
```

Comment:
As commented in a recent PR, I personally find misleading returning 200 via an `http.Error()`. We could simply add a `util.WriteTextResponse()` helper (similarly to the already existing `WriteJSONResponse()`) in order to preserve the 1 line to write both response content and response code.